### ISSUE TITLE
A11y - Pagination select has no accessible name

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#3734]
## Issue Description

Ensures select element has an accessible name

## Element Location

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c2b51159237ca50bb903e3017c6663f0a3a1dcdd/frontend/common/src/components/Pagination/Pagination.tsx#L205-L226

## To solve this issue, you need to...

Fix at least one (1) of these issues:

- Form element does not have an implicit (wrapped) <label>
- Form element does not have an explicit <label>
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Element has no title attribute
- Element's default semantics were not overridden with role="none" or role="presentation"
